developing/on-call/README.md

```diff
@@ -0,0 +1,193 @@
+# [Tools and Practice](../README.md) / On-Call Best Practices
+
+## Overview
+
+In a modern development environment, we want to make sure that the people
+writing the code own the services they write in production. Part of that
+ownership is sharing the burden of the on-call rotation. In order to make
+sure that said burden is not too arduous for everyone involved, here are
+some best practices when developing the on-call practice for your project.
+
+## Assumptions
+
+For the sake of argument, most of this article assumes you will be using
+[PagerDuty](https://www.pagerduty.com) for handling the actual alerting
+of the engineers on-call; if you are using another provider, most of these
+recommendations can be adapted for them. You can read more about various
+alerting providers in the [Alert Providers](../../infra/alerting/alert-providers.md)
+guide.
+
+## On-Call Rotations
+
+An on-call rotation consists of a pool of engineers who share a schedule
+that determines who is on-call at any one time.
+
+* For a single on-call shift, you should have a primary responder, a
+  secondary responder, and a tertiary "backstop", usually a lead, as the
+  final link in the chain. During the shift, the primary is expected to
+  respond to all alerts; the secondary and tertiary are there if for some
+  reason they are unable to respond (see [Escalation and Notification
+  Policies](./README.md#escalation-and-notification-policies) to see how
+  this is accomplished).
+* The secondary and tertiary also exist as additional resources for the
+  primary to call in as first points of contact for assistance if they
+  have a particularly bad or difficult incident. They can help diagnose
+  or remediate issues, contact subject matter experts for assistance, or
+  handle the logistics of incident response if necessary.
+* In most cases, you should aim for 6-8 people in the pool for a rotation.
+  This allows a schedule that maximizes the time you are not on-call,
+  while still having the rotation frequent enough that knowledge does not
+  become stale. In almost no case should there be fewer than 4 people nor
+  more than 12 people in a rotation pool. A pool of 4 or fewer people
```

Comment:
12 feels high. Maybe let's just stick to +/- 2 from your 6-8 proposal. So 4-10 as acceptable range.

Reply:
I used 12 as the upper limit for a couple of reasons. One, rounds out one week of primary and one week of secondary per three-month period. Two, you can split it evenly in half and have 6 people (which is the lower end of the recommended range.

Reply:
I think that's fair.